internal/http/middleware/authorization/kessel.go

```diff
@@ -0,0 +1,162 @@
+package authorization
+
+import (
+	"config-manager/internal/config"
+	"config-manager/internal/instrumentation"
+	"context"
+	"errors"
+	"net/http"
+
+	kesselv2 "github.com/project-kessel/inventory-api/api/kessel/inventory/v1beta2"
+	"github.com/project-kessel/inventory-client-go/common"
+	v1beta2 "github.com/project-kessel/inventory-client-go/v1beta2"
+	"github.com/redhatinsights/platform-go-middlewares/v2/identity"
+	"google.golang.org/grpc"
+)
+
+func NewKesselClient(config config.Config) KesselMiddlewareBuilder {
+	options := []func(*common.Config){
+		common.WithgRPCUrl(config.KesselURL),
```

Comment:
What is the criteria to select RPC over HTTP?

Reply:
https://docs.google.com/document/d/1KkVii7dU-p86rkj5HA8RXcI4Y3WcMgcoaIFpcigWP3I/edit?tab=t.0#heading=h.pfjzbznm9cbk